File: Max-Consecutive-Ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let maxCount = 0, currentCount = 0;
    for (const num of nums) {
        if (num === 1) {
            currentCount++;
            maxCount = Math.max(maxCount, currentCount);
        } else {
            currentCount = 0;
        }
    }
    return maxCount;
};

File: Max-Consecutive-Ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        maxCount = 0
        currentCount = 0
        for num in nums:
            if num == 1:
                currentCount += 1
                maxCount = max(maxCount, currentCount)
            else:
                currentCount = 0
        return maxCount
